torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -984,9 +986,14 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
                 << " (strides = " << output.strides() << ")" << std::endl;
     }
     std::cout << "Reduction and semaphore buffers:" << std::endl;
-    for (const auto& buffer : global_buffers.buffers) {
+    TORCH_INTERNAL_ASSERT(
+        global_buffers.buffers.size() == global_buffers.zero_init.size(),
```

Comment:
This is likely true in practice but not strictly true of the system. We could always set an intermediate buffer to be located in global memory.

Reply:
Are you just saying that we could have global buffer that's not zero init?

`global_buffers.zero_init` is a `vector<bool>`, which already considered that.